### PR TITLE
Feature/8631- Update fees on Manual transaction records

### PR DESCRIPTION
**Pull request by karthik-aot**
*Issue #:* /bcgov/entity/8631

*Description of changes:*

- Integration with fee service on passing filing type/quantity/priority
- updated transaction component
- test cases

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/composables/ViewRoutingSlip/useAddManualTransactionDetails.ts [hunk 1]
```diff
@@ -1,68 +1,71 @@
...
-  }
+  // vuex action and state
+  const { getFeeByCorpTypeAndFilingType } = useActions([
+    'getFeeByCorpTypeAndFilingType'
+  ])
 
-  function addManualTransactionRow () {
-    manualTransactionsList.value.push(getDefaultRow())
+  // Calculate total fee from pay-api service, triggered if its dependent values are changed
+  async function calculateTotal () {
```
Comment (karthik-aot) [inline src/composables/ViewRoutingSlip/useAddManualTransactionDetails.ts:28]: We can weigh in between this or using a watch

File: src/components/ViewRoutingSlip/AddManualTransactionDetails.vue [hunk 2]
```diff
@@ -1,132 +1,114 @@
...
+        v-model="manualTransactionDetails.quantity"
+        type="number"
+        @keyup="delayedCalculateTotal()"
+      >
+      </v-text-field>
+    </v-col>
+    <v-col cols="5">
+      <v-text-field
+        filled
+        label="Incorporation/Reference Number"
```
Comment (saravanpa-aot) [inline src/components/ViewRoutingSlip/AddManualTransactionDetails.vue:29]: i think referenceNumber is not mandatory rite?

File: src/components/ViewRoutingSlip/RoutingSlipTransaction.vue [hunk 3]
```diff
@@ -89,22 +108,34 @@ import { useRoutingSlipTransaction } from '@/composables/ViewRoutingSlip'
...
+      isDividerVisible,
+      isValid,
+      removeManualTransactionRow,
+      updateManualTransactionDetails
     }
   }
 })
-export default class RoutingSlipTransaction extends Vue {}
+export default class RoutingSlipTransaction extends Vue {
+  public getIndexedTag (tag, index): string {
```
Comment (saravanpa-aot) [inline src/components/ViewRoutingSlip/RoutingSlipTransaction.vue:137]: move this to somewhere plz ..this getIndexedtag ;may be a mixin

**Review by saravanpa-aot — APPROVED** [review]

File: src/components/ViewRoutingSlip/AddManualTransactionDetails.vue [hunk 4]
```diff
@@ -1,132 +1,114 @@
...
+      delayedCalculateTotal
     }
   }
 })
 export default class AddManualTransactionDetails extends Vue {
-  @Prop({ default: () => [] }) value: ManualTransactionDetails[]
+  @Prop({ default: () => undefined }) index: number
+  @Prop({ default: () => null }) manualTransaction: ManualTransactionDetails
 
   public getIndexedTag (tag, index): string {
```
Comment (shabeeb-aot) [inline src/components/ViewRoutingSlip/AddManualTransactionDetails.vue:113]: move this to composable functions

File: src/composables/ViewRoutingSlip/useRoutingSlipTransaction.ts [hunk 5]
```diff
@@ -1,35 +1,74 @@
...
-  const addManualTransactionsList = ref<ManualTransactionDetails[]>([])
+  const formRoutingSlipManualTransactions = ref<HTMLFormElement>()
+  const manualTransactionsList = ref<ManualTransactionDetails[]>([])
 
   function showManualTransaction (): void {
-    // show manual transaction component through toggling showAddManualTransaction
+    // Show manual transaction component through toggling showAddManualTransaction
     showAddManualTransaction.value = !showAddManualTransaction.value
+    // And add a default row to the manual transaction list
+    addManualTransactionRow()
```
Comment (shabeeb-aot) [inline src/composables/ViewRoutingSlip/useRoutingSlipTransaction.ts:15]: need to add if empty

File: src/composables/ViewRoutingSlip/useRoutingSlipTransaction.ts [hunk 6]
```diff
@@ -1,35 +1,74 @@
...
-    addManualTransactionDetailsRef.value.addManualTransactionRow()
+  function addManualTransactions (): void {
+    if (isValid()) {
+    }
   }
 
-  function addManualTransactions (): void {
-    if (addManualTransactionDetailsRef.value.isValid()) {
+  // Divider not visible if array is 1 OR last array element
+  function isDividerVisible (index: number): boolean {
```
Comment (shabeeb-aot) [inline src/composables/ViewRoutingSlip/useRoutingSlipTransaction.ts:24]: isLastChild?

File: src/composables/ViewRoutingSlip/useAddManualTransactionDetails.ts [hunk 7]
```diff
@@ -1,68 +1,71 @@
...
+        }
+        nextTick(async () => {
+          manualTransactionDetails.value.total = await getFeeByCorpTypeAndFilingType(getFeeRequestParams)
+        })
+      } else {
+        manualTransactionDetails.value.total = null
+      }
+    } catch (error) {
+      // TODO : Business errors (400s) need to be handled
+      // eslint-disable-next-line no-console
```
Comment (shabeeb-aot) [inline src/composables/ViewRoutingSlip/useAddManualTransactionDetails.ts:48]: manualTransactionDetails.value.total = null is needed here also?

**Review by shabeeb-aot — APPROVED** [review]

File: src/components/ViewRoutingSlip/AddManualTransactionDetails.vue (hunk 2 above)
Comment (karthik-aot) [reply src/components/ViewRoutingSlip/AddManualTransactionDetails.vue:29]: no. It is required 
image

**Pull request merged**